test/systeminfotest/SystemInfoTest.scala

```diff
@@ -1,20 +1,51 @@
+/*
+ * Copyright (C) 2018 The Delphi Team.
+ * See the LICENCE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package systeminfotest
 
-import org.scalatest.FlatSpec
-import controllers._
+import controllers.SystemInfoController
+import models.SystemInfo
 
-class SystemInfoTest  extends FlatSpec{
-  it should "contains non empty java version" in {
-    assertResult(false)(Option(JavaVersion(None)).isEmpty)
-  }
-  it should "contains non empty scala version" in {
-    assertResult(false)(Option(ScalaVersion(None)).isEmpty)
-  }
-  it should "contains non empty hostname" in {
-    assertResult(false)(Option(HostName(None)).isEmpty)
-  }
-  it should "contains non empty platform value" in {
-    assertResult(false)(Option(JavaVersion(None)).isEmpty)
+import scala.concurrent.Future
+import org.scalatestplus.play._
+import play.api.libs.json._
+import play.api.mvc._
+import play.api.test._
+import play.api.test.Helpers._
+
+
+class SystemInfoTest extends PlaySpec with Results {
+
+  "SystemInfo" should {
+    "should return a valid SystemInfo Json" in {
+      val controller = new SystemInfoController(stubControllerComponents())
+
+      val result: Future[Result] = controller.getInfo().apply(FakeRequest())
+      val bodyText: String = contentAsString(result)
+      val json = Json.parse(bodyText)
+      implicit val systemInfoWrites = Json.writes[SystemInfo]
+      implicit val systemInfoReads = Json.reads[SystemInfo]
+
+      val validateResult: JsResult[SystemInfo] = json.validate[SystemInfo]
+
+      validateResult.isSuccess mustBe true
```

Comment:
That is a very very basic test. Could you extend it?

Reply:
Admittedly, I'm not sure how. 
The content of the received json object is depending on the machine, which runs the test.
Therefore, if I want to do more than compare the structure of the object I would have to query these information in the test case dynamically, in the same way I query them in the method I'm testing. Then these result would always match anyways. 
The only advantage I see here is that the test case might break if I
we change the code to something new by mistake maybe. 

If you have any further suggestions please state them, this is actually a problem I have thought about for quite some time when writing the current test.

Reply:
Mhh... That's quite a good justification. Let's keep that for now.